### ISSUE TITLE
No RAs on dummy device with "::/64"

### ISSUE THREAD

**Issue by ivilata** [reubenhwk/radvd#46]
Hi, I have a Debian Sid setup with a dummy interface which is manually configured with a deprecated, fixed address before radvd is started, so that the daemon can use a `::/64` prefix and announce on that interface.  Autoconfiguration is then used in the interface to get temporary privacy addresses.  This is the relevant stanza in `/etc/network/interfaces`:

```
auto dummy0
iface dummy0 inet6 auto
    pre-up  ip addr add W:X:Y:Z::1/64 dev $IFACE preferred_lft 0
    privext  2
```

And this is the full `/etc/radvd.conf`:

```
interface dummy0 {
    AdvSendAdvert on;
    prefix ::/64 {};
};
```

This used to work perfectly with radvd 1.9.1 (1.9.1-1.3 from Debian), but with 2.11 (2.11-1) the interface never gets temporary addresses.  Using `radvdump` and a sniffer I see that RAs are sent on `dummy0` with 1.9.1, but no RAs are seen with 2.11.

I have tried enabling `UnicastOnly` (since radvd complains that the interface doesn't support multicast) and `IgnoreIfMissing` just in case, but no luck.  Are there any other options I can test if the behaviour has changed from 1.x?  Thanks!

**Comment by ivilata** [comment]
BTW, I see an entry in the changelog:

    2014/05/30	add disable_ipv6_autoconfig function so an interface radvd
                		is using won't autoconfig itself using its own advert

I found no such function in the current code, maybe it was removed or somehow rewritten later.

**Comment by reubenhwk** [comment]
I'm not sure.  I'll look into it when I get a chance.

**Comment by ivilata** [comment]
Hi!  After more testing, I think I found the reason why this doesn't work.  I enabled debugging with version 2.13 and saw the following message (as I mentioned above):

> dummy0 does not support multicast, forcing UnicastOnly

which I traced to ``send.c:send_ra_forall()``, and also this new message:

> no client list, no destination, unicast only...doing nothing

which I traced to ``device-common.c:check_device()``.  Manually disabling the conditions that fired these messages and rebuilding resulted in radvd sending announcements on ``dummy0``.

I was somehow confused since multicast pings like ``ping6 ff02::1%dummy0`` worked as expected, but I realised in the output of ``ifconfig dummy0`` (``<UP,BROADCAST,RUNNING,NOARP>``) that the ``MULTICAST`` flag wasn't enabled (maybe it's the default behaviour for Linux).  After deconfiguring ``dummy0`` and stopping radvd, running ``ip link set multicast on dev dummy0``, and then putting ``dummy0`` back up and starting the (not modified) radvd, it had no problem in sending announcements to that interface.

As I mentioned in the first comment, this setup worked with radvd 1.9.1 without setting the multicast flag, but maybe its checks were too weak and the new behaviour is actually intended, so feel free to close the issue. I'm glad anyway to have solved the mystery finally.`:)`

Thanks!
